XamarinCommunityToolkit/Views/CameraView/CameraViewRenderer.uwp.cs

```diff
@@ -0,0 +1,397 @@
+﻿using System;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Windows.Devices.Enumeration;
+using Windows.Devices.Lights;
+using Windows.Graphics.Display;
+using Windows.Graphics.Imaging;
+using Windows.Media.Capture;
+using Windows.Media.Core;
+using Windows.Media.Devices;
+using Windows.Media.MediaProperties;
+using Windows.Storage;
+using Windows.Storage.Streams;
+using Windows.System.Display;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Xamarin.CommunityToolkit.UI.Views;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.UWP;
+using MediaSource = Xamarin.Forms.MediaSource;
+
+[assembly: ExportRenderer(typeof(CameraView), typeof(CameraViewRenderer))]
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+	public class CameraViewRenderer : ViewRenderer<CameraView, CaptureElement>
+	{
+		MediaCapture mediaCapture;
+		bool isPreviewing;
+		Lamp flash;
+		DisplayRequest displayRequest = new DisplayRequest();
+		LowLagMediaRecording mediaRecording;
+		string filePath;
+		bool busy;
+		VideoStabilizationEffect videoStabilizationEffect;
+		MediaEncodingProfile encodingProfile;
+		VideoEncodingProperties inputPropertiesBackup;
+		VideoEncodingProperties outputPropertiesBackup;
+
+		public CameraViewRenderer()
+		{
+			encodingProfile = MediaEncodingProfile.CreateMp4(VideoEncodingQuality.Auto);
+		}
+
+		bool IsBusy
+		{
+			get => busy;
+			set
+			{
+				if (busy != value)
+					Element.IsBusy = busy = value;
+			}
+		}
+
+		bool Available
+		{
+			get => Element?.IsAvailable ?? false;
+			set
+			{
+				if (Element != null && Element.IsAvailable != value)
+					Element.IsAvailable = value;
+			}
+		}
+
+		protected override async void OnElementChanged(ElementChangedEventArgs<CameraView> e)
+		{
+			Available = false;
+			base.OnElementChanged(e);
+			if (e.OldElement != null)
+			{
+				e.OldElement.ShutterClicked += HandleShutter;
+			}
+			if (e.NewElement != null)
+			{
+				if (Control != null)
+				{
+					await CleanupCameraAsync();
+					mediaCapture.Failed -= MediaCaptureFailed;
+				}
+
+				SetNativeControl(new CaptureElement());
+				Control.HorizontalAlignment = HorizontalAlignment.Stretch;
+				Control.VerticalAlignment = VerticalAlignment.Stretch;
+
+				e.NewElement.ShutterClicked += HandleShutter;
+
+				isPreviewing = false;
+				await InitializeCameraAsync();
+
+				if (mediaCapture != null)
+					mediaCapture.Failed += MediaCaptureFailed;
+			}
+		}
+
+		async void HandleShutter(object sender, EventArgs e)
+		{
+			if (IsBusy)
+				return;
+
+			IsBusy = true;
+			switch (Element.CaptureOptions)
+			{
+				default:
+				case CameraCaptureOptions.Default:
+				case CameraCaptureOptions.Photo:
+					if (mediaRecording != null)
+						goto case CameraCaptureOptions.Video;
```

Comment:
`go to` LoL

Reply:
Lol, yeah I noticed that too, I didn't even know you could do that!

Reply:
Me neither kkkk

Reply:
Brrrr unfortunately I know you can... Can we get that out?

Reply:
I'm sure it could be refactored to not use GoTo, I'll take a look

Reply:
Replaced with a local function called inside the various case statements